### PR TITLE
Implement Entry IsSpellCheckEnabled property

### LINKED ISSUE #8558: Entry.IsSpellCheckEnabled="False" still shows squiggly line
### Description

I have an entry field with IsSpellCheckEnabled="False". When text is entered into the field and it fails the spellchecker, it still draws a squiggly line under the text. I am expecting no squiggly line.

Screenshot 2022-07-06 071101

This is the xaml used to define the entry control:

```
<Entry
	x:Name="EmailAddressEntry"
	ClearButtonVisibility="WhileEditing"
	IsSpellCheckEnabled="False"
	Margin="0,2"
	Placeholder="{x:Static strings:AppResources.PlaceholderEnterEmailAddress}"
	Text="{Binding EmailAddress}" />
```

This field is used to enter an email address so there's a 99.9999% chance it will fail a spellchecker...

### Steps to Reproduce

1) In ANY Maui app, add an entry field and use the following definition:
```
<Entry
	x:Name="EmailAddressEntry"
	ClearButtonVisibility="WhileEditing"
	IsSpellCheckEnabled="False"
	Margin="0,2"
	Placeholder="{x:Static strings:AppResources.PlaceholderEnterEmailAddress}"
	Text="{Binding EmailAddress}" />
```
2) Enter "asdefasdf@asdasdf.com"
3) The dreaded squiggly will be drawn showing the "spelling error"

### Version with bug

6.0.400 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

windows10.0.19041.0

### Did you find any workaround?

NO

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Implement Entry `IsSpellCheckEnabled `property.

Fixes (but reverted) #8558

### CODE REVIEW

**Comment by rmarinho** [comment]
@jsuarezruiz  any idea why tests are failing ?

**Review by mattleibow — APPROVED** [review]
Looks good, just the API breaks are gonna be fun.

**Comment by rmarinho** [comment]
@jsuarezruiz  can you add this to the API ?

**Comment by rmarinho** [comment]
@jsuarezruiz can you rebase this one ? i might have screwed it up

**Comment by jsuarezruiz** [comment]
To help focus and move faster, we're going to have fewer open PRs and focus on the ones we'll be working on in each release. This PR is not included in the next release so I close it for now. I will reopen it and adapt to changes later.

**Review by rmarinho — APPROVED** [review]

**Review by mandel-macaque — CHANGES_REQUESTED** [review]
Minor nit, but I think using HasFlag will make the coder  ore readable and accessible to more jr devs.

File: src/Core/src/Platform/Android/EditTextExtensions.cs [hunk 1]
```diff
@@ -288,8 +293,17 @@ internal static void SetInputType(this EditText editText, ITextInput textInput)
 
 				if ((nativeInputTypeToUpdate & InputTypes.TextFlagNoSuggestions) != InputTypes.TextFlagNoSuggestions)
 				{
-					if (!textInput.IsTextPredictionEnabled)
-						nativeInputTypeToUpdate |= InputTypes.TextFlagNoSuggestions;
+					if ((nativeInputTypeToUpdate & InputTypes.TextFlagNoSuggestions) != InputTypes.TextFlagNoSuggestions)
```
[the remark this answers is not in the document]
Comment (jsuarezruiz) [reply src/Core/src/Platform/Android/EditTextExtensions.cs]: Changed.

**Review by mandel-macaque — APPROVED** [review]

**Comment by mattleibow** [comment]
This was reverted in https://github.com/dotnet/maui/pull/14147

**Pull request merged**